component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -166,134 +188,161 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
-                    authenticatedUser = stepConfig.getAuthenticatedUser();
-                    if (authenticatedUser != null && stepConfig.isSubjectAttributeStep()) {
-                        username = authenticatedUser.toFullQualifiedUsername();
-                        if (stepConfig.getAuthenticatedIdP().equals(EmailOTPAuthenticatorConstants
-                                .LOCAL_AUTHENTICATOR)) {
-                            isLocalUser = true;
+        if (context.getLastAuthenticatedUser() == null) {
+            context.setProperty(EmailOTPAuthenticatorConstants.IS_IDF_INITIATED_FROM_AUTHENTICATOR, true);
+            String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL();
+            String queryParams = context.getContextIdIncludedQueryParams();
+            try {
+                if (log.isDebugEnabled()) {
+                    String logMsg = String.format("Redirecting to identifier first flow since " +
+                                    "last authenticated user is null in SP: %s",
+                            context.getServiceProviderName());
+                    log.debug(logMsg);
+                }
+                // Redirecting the user to the login page
+                response.sendRedirect(loginPage + ("?" + queryParams) + "&" + EmailOTPAuthenticatorConstants.AUTHENTICATORS
+                        + EmailOTPAuthenticatorConstants.IDF_HANDLER_NAME + ":" + EmailOTPAuthenticatorConstants.LOCAL_AUTHENTICATOR);
+            } catch (IOException e) {
+                User user = User.getUserFromUserName(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME));
+                throw new AuthenticationFailedException(
+                        EmailOTPAuthErrorConstants.ErrorMessages.SYSTEM_ERROR_WHILE_AUTHENTICATING.getCode(),
+                        e.getMessage(), user, e);
+            }
+        } else {
+            try {
+                boolean isEmailOTPMandatory;
+                boolean sendOtpToFederatedEmail;
+                String usecase;
+                Object propertiesFromLocal = null;
+                String email;
+                AuthenticatedUser authenticatedUser = null;
+                Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+                String tenantDomain = context.getTenantDomain();
+                context.setProperty(EmailOTPAuthenticatorConstants.AUTHENTICATION,
+                        EmailOTPAuthenticatorConstants.AUTHENTICATOR_NAME);
+                if (!tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
+                    IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, getName(), tenantDomain);
+                    propertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+                }
+                if (propertiesFromLocal != null || tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
```

Comment:
Use `constant.equals(variable)` rather than `variable.equals(constant)` to avoid any *NullPointerException*.

Reply:
Addressed